### ISSUE TITLE
2FA login page & reverse proxy

### ISSUE THREAD

**Issue by jtrosper** [webmin/webmin#1630]
Hello,

I am running Webmin behind an Nginx reverse proxy.  I have `webprefixnoredir=1` defined.  Once that directive was in place, Webmin no longer appended :10000 and I was able to login without issues.

However, after enabling 2FA, the 2FA login page itself seems to ignore the webprefixnoredir directive and is still appending 10000.

By simply deleting the :10000 from the address bar, the session appears to be properly authenticated and I can get in.

**Comment by jcameron** [comment]
Can you post a screenshot of this? I'm not sure what you mean by the 2FA login page.

**Comment by jtrosper** [comment]
[Screenshot one - the page where you enter the 2FA token.](https://imgur.com/a/u24pXVv)

[Screenshot two - :10000 being appended after authenticating the session, despite webprefixnoredir=1 being defined.](https://imgur.com/a/nmVQoGc)

Deleting the port from the URL and hitting enter then takes me directly into my authenticated session.

**Comment by jcameron** [comment]
@iliaross can you take a look at this? I assume this is due to theme changes to the 2fa field on the login form.

**Comment by iliajie** [comment]
Yes, I'll have a look tomorrow.

**Comment by iliajie** [comment]
@jcameron Although, if I remember correctly the redirect is done by _miniserv_, so it shouldn't be the theme issue.

@jtrosper If you run Webmin 1.991, check _Webmin Configuration/Webserver Options_ page. You should try setting post-login redirects there. Also, this question has been asked and answered before.

**Comment by jcameron** [comment]
@iliaross the Authentic theme modifies the sign-in page to add an extra step when the user has 2fa enabled though, right?

**Comment by jtrosper** [comment]
@iliaross @jcameron and anyone else who may stumble upon this in the future.  I was able to resolve this issue, though I don't know if this is the correct solution, or just a functional one.

I had already defined the following in /etc/webmin/config:
```
webprefixnoredir=1
referers=my.proxy.address
```

Per the suggestion of a years-old post on StackOverflow, I added the following:
`relative_redir=0`

Next, I already had the following defined in /etc/webmin/miniserv.conf:
`ssl=0` (since the proxy is handling SSL for hosts behind it)

Per a suggestion from another years-old closed issue here on Github, I added this:
`redirect_port=443`

The combination of all of the above directives seems to have resulted in the desired behavior of stripping port 10000 from any redirection, including the 2FA authentication page.

And @iliaross - Webmin is literally decades old, and so are discussions, issues, and forum posts about its functionality.  Saying "this question has been asked and answered before" has a net benefit of zero, and is also peak FOSS developer snide commentary.

I'm closing this issue now.  Good luck to future Googlers.
